### ISSUE TITLE
Enable Ribbon button on Excel document load

### ISSUE THREAD

**Issue by savanetllc** [OfficeDev/office-js#2375]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

We are deveoping Excel JS Addin and have couple of Ribbon buttons, few of them are disabled from manifest file (those are Excel Template specific buttons). So if specific Excel template document (which have specific sheets) opened then we need to enable those ribbon buttons. Currently everything works fine if user first click on ribbon button which open taskpane so disabled buttons also got enabled fine. We wrote enable button functions in `Excel.onReady` and we need to enable button as soon as Excel template document open (not after Taskpane open). 
So is there any option to trigger enable button routine in JS Addin at document opening time? Thank you.

**Comment by RuizhiSunMS** [comment]
Thanks for your question @savanetllc, here are some questions we want to ask to make the scenario clear:
1. for now those disabled buttons could only be enabled after task pane opened?
2. By which ways do you expect to implement expected functions? Via manifest or by some apis?
3. Is there any another workaround you could accept?

**Comment by savanetllc** [comment]
Hi @RuizhiSunMS Thanks for your reply. Here are answers of each question in order.

1. Yes those buttons enabled after task pane opened. 
2. I am looking for possible ways to meet with mentioned requirements Any way to enable ribbon buttons without opening task pane is acceptable. 
3. Yes, sure.

Let me rephrase my requirements for better understanding. We have 4 ribbon buttons in Excel in which 3 buttons are disabled in manifest file using following element `<Control xsi:type="Button" id="addEstimateYearButton">.......<Enabled>false</Enabled><Action xsi:type="ExecuteFunction"><FunctionName>addEstimateYear</FunctionName></Action></Control>` and one button is enabled with action `ShowTaskpane`. So these three buttons will remain disabled for all Excel document until if opening Excel Document has three work sheets with name "Sheet1", "Sheet2" and "Sheet3". Only for this document these buttons will be enabled otherwise remain disabled. (This specific Excel document is exported from other application so it will have specific Worksheets with name and these buttons will work only this documents).

Currently on `Office.onReady()` I check opened document worksheets and then enable buttons using `Office.ribbon.requestUpdate` but this only work when user open Taskpane by pressing its (enabled) button but I am looking for to enable buttons wihtout opening Task pane because Task pane UI is for another function (not related to these three buttons functionality).

Hope it clears.

**Comment by AbidRahman-MSFT** [comment]
Hey @savanetllc - is your add-in utilizing shared runtime? Here is more details on the shared runtime feature: https://docs.microsoft.com/en-us/office/dev/add-ins/design/disable-add-in-commands#shared-runtime-required

**Comment by trigurl** [comment]
@AbidRahman-MSFT, in order to use the Enable/Disable API, it has to be shared runtime.

Hi @savanetllc

So if you want this to work without any UI, then you need to build a shared runtime add-in, then put your add-in in a template to get a UI-less add-in to run on file open. 
https://learn.microsoft.com/en-us/office/dev/add-ins/develop/run-code-on-document-open
However, not all versions of add-ins support shared runtime.  If you need to run in versions of Office that do no support shared runtime, then you need to use the auto open task pane feature that will auto open the task pane on file open.

[https://learn.microsoft.com/en-us/office/dev/add-ins/develop/automatically-open-a-task-pane-with-a-document](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Foffice%2Fdev%2Fadd-ins%2Fdevelop%2Fautomatically-open-a-task-pane-with-a-document&data=05%7C01%7Cangelaso%40microsoft.com%7C5a162f13fabe4155ef7408db19f07d52%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638132292912896114%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=qg5Tn8gy4DbmY0Qt3x1%2Bj9CYJhXhzwEmzlI4R7luuF4%3D&reserved=0)
